encoding/thrift/transport_options.go

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package thrift
+
+import "github.com/yarpc/yarpc-go/transport"
+
+// DisableEnvelopingForTransport allows a transport to disable Thrift request
+// and response enveloping.
+//
+// Use this from a transport implementation that wishes to disable Thrift
+// enveloping.
+//
+// 	func (myOutbound) Options() transport.Options {
+// 		return otherOptions.Merge(thrift.DisableEnvelopingForTransport)
+// 	}
+var DisableEnvelopingForTransport transport.Options
+
+func init() {
+	DisableEnvelopingForTransport = DisableEnvelopingForTransport.With(
```

Comment:
nit: suggest `transport.Options{}.With(...)` which is a little clearer that `DisableEnvelopingForTransport` doesn't rely on any previous values